src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -294,6 +363,7 @@ def _parse_function_args(cls, func):
                      (v.kind == inspect.Parameter.POSITIONAL_ONLY or
                       v.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD)]
         else:
+            func = getattr(func, '__wrapped__', func)
```

Comment:
AFAIK, this only works in Python 3. I'm working on a solution that also works for python2

Reply:
this also works in python 2 because I'm explicitly adding that attribute in: https://github.com/ceph/ceph/pull/21360/files#diff-3095fd00dfad68aeeccd8ad3fe6d3356R327